src/deposit/views.py

```
@@ -38,6 +64,48 @@ def _verify_optional_args(request):
     return None
 
 
+@api_view(["GET", "POST"])
+def interactive_deposit(request):
+    # Validate query parameters: account, asset_code, transaction_id.
+    account = request.GET.get("account")
+    if not account:
+        return render_error_response("no 'account' provided")
+
+    asset_code = request.GET.get("asset_code")
+    if not asset_code or not Asset.objects.filter(name=asset_code).exists():
+        return render_error_response("invalid 'asset_code'")
+
+    transaction_id = request.GET.get("transaction_id")
+    if not transaction_id:
+        return render_error_response("no 'transaction_id' provided")
+
+    # POST: The user submitted a form with the amount to deposit.
+    if request.method == "POST":
+        form = DepositForm(request.POST)
+        form.asset = Asset.objects.get(name=asset_code)
```

Right now, if a Transaction with the given `id` already exists, the server will return a 500 response, which means we're not handling that exception properly.

I think the correct flow would be:
1. Check if a `Transaction` with the given `id` already exists
2. If it does, we should return an error message (something in the lines of `{ "error": "transaction id already in use" }` with a 400 (Bad Request) status code.
3. If it doesn't, we proceed with the flow you've defined on the following `if`.

---

Ah, got it. Sorry, I misunderstood the failure case before - implementing this flow now.

---

Oh, you actually made the correct changes – this is just a remaining edge case that still hasn't been covered :)